api/src/main/java/io/druid/segment/loading/DataSegmentPusherUtil.java

```diff
@@ -64,4 +64,21 @@ public static String getHdfsStorageDir(DataSegment segment)
         segment.getShardSpec().getPartitionNum()
     );
   }
+
+  /**
+   * Due to https://issues.apache.org/jira/browse/HDFS-13 ":" are not allowed in
+   * path names. So we format paths differently for HDFS.
+   */
+  public static String getHdfsStorageDirUptoVersion(DataSegment segment)
```

Comment:
minor nit : in  getHdfsStorageDir() method above, you can actually reUse this method.